foundation/storage/src/test/kotlin/com/pingidentity/storage/DataStorageTest.kt

```diff
@@ -7,44 +7,86 @@
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
 import kotlinx.coroutines.test.runTest
 import org.junit.Rule
+import org.junit.rules.TestWatcher
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
+import java.io.File
 import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
 import kotlin.test.Test
 import kotlin.test.assertEquals
-import org.junit.rules.TestWatcher
 
 @RunWith(RobolectricTestRunner::class)
 class DataStorageTest {
     @JvmField
     @Rule
     val watcher: TestWatcher = TestRailWatcher
 
-    private val context: Context by lazy { ApplicationProvider.getApplicationContext<Application>() }
+    private val testContext: Context = ApplicationProvider.getApplicationContext()
+    private val testCoroutineScheduler = TestCoroutineScheduler()
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val testScope = TestScope(UnconfinedTestDispatcher(testCoroutineScheduler))
+
+    private lateinit var dataStore: DataStore<Data?>
+    private lateinit var dataStoreList: DataStore<List<Data>?>
+    private lateinit var dataStoreFile: File
+    private lateinit var dataStoreListFile: File
+
+    @BeforeTest
+    fun setup() {
+        // Use a unique file name for each test to ensure isolation
+        val fileName = "test_data_${System.currentTimeMillis()}.pb"
+        dataStoreFile = testContext.dataStoreFile(fileName)
+
+        dataStore = DataStoreFactory.create(
+            serializer = DataToJsonSerializer(), // Your actual serializer
+            scope = testScope, // Use the test scope
+            produceFile = { dataStoreFile }
+        )
+
+        val fileNameList = "test_datalist_${System.currentTimeMillis()}.pb"
+        dataStoreListFile = testContext.dataStoreFile(fileNameList)
 
-    private val Context.dataStore: DataStore<Data?> by dataStore("test", DataToJsonSerializer())
-    private val Context.dataStoreList: DataStore<List<Data>?> by dataStore("test-list", DataToJsonSerializer())
+        dataStoreList = DataStoreFactory.create(
+            serializer = DataToJsonSerializer(), // Your actual serializer
+            scope = testScope, // Use the test scope
+            produceFile = { dataStoreListFile }
+        )
+
+
+    }
 
     @AfterTest
-    fun tearDown() =
-        runTest {
-            context.dataStore.updateData { null }
+    fun tearDown() {
+        // Crucially, delete the DataStore file after each test
+        if (dataStoreFile.exists()) {
+            dataStoreFile.delete()
         }
 
+        if (dataStoreListFile.exists()) {
+            dataStoreListFile.delete()
+        }
+    }
+
     @TestRailCase(21605, 21611)
     @Test
     fun testDataStore() =
         runTest {
-            val storage = DataStoreStorage(context.dataStore)
+            val storage = DataStoreStorage(dataStore)
+            //val storage = DataStoreStorageFactory.getOrCreate<Data>("test")
```

Comment:
Is that code intentionally commented out? 

> //val storage = DataStoreStorageFactory.getOrCreate<Data>("test")

Reply:
The commented out line is removed.